test/utils/pickers/fields.tsx

```diff
@@ -173,6 +179,34 @@ export const buildFieldInteractions = <P extends {}>({
       });
     };
 
+    const selectSectionAsync: FieldSectionSelectorAsync = async (
```

Comment:
Added this because the `act`s below would sometimes fail without being async. Not sure if it would be better to change the original `selectSection` to be async instead. Maybe in a future step?

Reply:
What would be the downside to have all the `selectSection` be async?
If we want to go in that direction, could you deprecate the old one and add a small JSDoc so that we rembember to keep working on it?

Reply:
Mainly a lot of file changes because the test functions will need to be made async.

Reply:
OK, I'm good with deprecating, adding a JSDoc and doing a follow up once the Vitest migration is done (to avoid Git conflicts)

Reply:
I agree with this, let's deprecate and add JSDoc. 👌